### PR TITLE
Video cropping + media elements refactor

### LINKED ISSUE #442: Video - Add features from Images
## Feature Description

This ticket is to add features built for images so they are also available for videos.

See image ticket: #35 
---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
1. The author is able to crop videos
2. Refactor video element to sync with Image (re-use code)

### PR DESCRIPTION
**Pull request by miina**
Fixes #442 

- Bring cropping (resizing in edit mode) to Video element.
- Reuses logic between media elements.

### CODE REVIEW

File: assets/src/edit-story/elements/media/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  &::after {
+    content: '';
+    display: block;
+    position: absolute;
+    left: 0;
+    top: 0;
+    width: 100%;
+    height: 100%;
+    border: 1px solid ${({ theme }) => theme.colors.mg.v1}70;
```
Comment (swissspidy) [inline assets/src/edit-story/elements/media/index.js]: I know this was just copy pasted, but what's with the 70 here? Opacity? We should probably use `rgba` for that.

File: assets/src/edit-story/elements/media/edit.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    width,
+    height,
+    scale,
+    flip?.horizontal ? 100 - focalX : focalX,
+    flip?.vertical ? 100 - focalY : focalY
+  );
+
+  mediaProps.transformFlip = getTransformFlip(flip);
+
+  const fadedMedia =
```
Comment (swissspidy) [inline assets/src/edit-story/elements/media/edit.js]: Why not name it `FadedMedia` and then use `<FadedMedia />` below?

File: assets/src/edit-story/elements/media/index.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/elements/media/index.js]: Hmm, seems like a typo to me, removed it.

**Comment by spacedmonkey** [comment]
@miina Unit tests are failing.

**Comment by miina** [comment]
@spacedmonkey Thanks, the tests are passing now.

**Review by spacedmonkey — APPROVED** [review]
👍

**Review by swissspidy — APPROVED** [review]

**Pull request merged**